src/vs/workbench/contrib/terminal/browser/terminalActions.ts

```diff
@@ -748,8 +748,11 @@ export function registerTerminalActions() {
 				precondition: TerminalContextKeys.processSupported
 			});
 		}
-		async run(accessor: ServicesAccessor) {
-			return accessor.get(ITerminalService).activeInstance?.changeIcon();
+		async run(accessor: ServicesAccessor, resource: unknown) {
+			const terminalService = accessor.get(ITerminalService);
+			const castedResource = URI.isUri(resource) ? resource : undefined;
+			const instance = terminalService.getInstanceFromResource(castedResource) || terminalService.activeInstance;
+			instance?.changeIcon();
```

Comment:
Could reduce some repetition by making something similar to `doWithActiveInstance`:

```ts
doWithResourceOrActiveInstance<T>(resource: unknown | URI | undefined, callback: (terminal: ITerminalInstance) => T): T | void {
}

async run(accessor: ServicesAccessor, resource: unknown) {
  accessor.get(ITerminalService).doWithResourceOrActiveInstance(e => e.changeIcon());
}
```

Reply:
thought about it, but wasn't sure if it was worth adding another method to terminal service since it's only used in 3 places.